### ISSUE TITLE
Deployed NFTs lose their metadata on both L1 and L2

### ISSUE THREAD

**Issue by sk33z3r** [Loopring/loopring-wallet-feedback#185]
See the latest update for newest information. TL;DR

### Users should not mint using the current UI until it is fixed

---

It's possible this is directly related to #184 , but I am posting separately since there are a lot of people reporting these symptoms. This also ties into #182.

**Repro**:
1. [Follow the steps at the official zoho link for minting NFTs to upload your content](https://desk.zoho.com/portal/loopring/en/kb/articles/nft-minting-on-l2)
2. Connect wallet to the web app using WalletConnect or MetaMask
3. Go to My NFTs
4. Click the Mint NFT button in top-right
5. Paste your metadata.json CID into the form
_OBSERVE_: Intermittent issues due to #186 
6. Click Mint My NFT and sign the approval message
7. Click on your newly minted NFT to view its details

Observations:
* Some minters still experience the behavior explained in #182 before deployment.
* Other minters can see their NFT fine before they deploy, but because of the confusion around #182 many people have defaulted to deploying their NFTs immediately, whether it looks OK or not.
* Minters who have minted in the past few days (before March 2nd) had already deployed their contracts. However, at this step they now see the Deploy Contract button again. This is due to #184 and the new smart contract, previous minters are now minting under a new contract address than previous.

**Issue**:
In all cases above, when someone deploys their contract, it appears the metadata disappears and the UI shows `--` for the name and description, as well as a placeholder image.

**Question for the Team**:
Is it possible this behavior is all directly related to the metadata change in the new smart contract, outlined in #184?

**Example of Minted NFT after Deployment**:
![image](https://user-images.githubusercontent.com/20587571/156591301-3ccaa7e6-874b-4ae0-a8de-10389756db36.png)

**Comment by sk33z3r** [comment]
I'd be curious, as well, if this ELI5 explanation is accurate:

* When the contract is on L2 only, it can dynamically find files within a CID for metadata.json
* When the contract is on L1 as well, L1 only resolves the CID and doesn't attempt to look for sub-files in a folder
* The minting UI forces you to put metadata.json into a folder when minting. So before you deploy, it works most of the time cause L2 can generally pull up the metadata. After deploy, it's broken because now L1 is involved and only resolves just the CID, and subsequently can't pull the metadata

**Comment by discombobulater** [comment]
In my case, I waited a day or so for the NFTs to be visible in L2 wallet before I hit the Deploy button.  Then I waited a couple hours for those transactions to complete before I clicked the Withdraw button using my L1 ETH address.  When that completed is when I saw null metadata even after waiting a day.  I can provide transaction IDs in a PM if it is useful to whoever is working this issue.

**Comment by sk33z3r** [comment]
Last observation: The "Unidentified Contract" and missing collection descriptions appear to be an OpenSea thing. Those are meant to be setup in the OpenSea interface, and are not taken from the `metadata.json`.

**Comment by discombobulater** [comment]
Here is what an NFT withdrawn from L2 to L1 looks like in Etherscan:

<img width="1404" alt="Screen Shot 2022-03-08 at 6 47 42 AM" src="https://user-images.githubusercontent.com/20411570/157262349-10a191ce-c3a6-4f5a-9fd6-ed951ffebbbf.png">

**Comment by sk33z3r** [comment]
We've had a couple of days now of minting being back online with a new UI. Users no longer have to name their json file `metadata.json` and should upload the file to pinata, instead of a folder with the file in it.

We confirmed that after deployment to L1, the contract and NFT make it to marketplaces like OpenSea without issue, and display data as expected.

Any issues with NFTs minted prior to the button being cut-off should be reported in the Discord channels support or bugs, and someone will be able to sort out the exact issue to resolve any problems, or determine if something was wrong with the mint (like incorrect json content, unpinned files in ipfs, etc.) that would mean a remint is needed. Most issues related to the contract switch have more or less been addressed with the new mint UI.

Officially closing this ticket.
